source/agora/node/Ledger.d

```diff
@@ -247,7 +254,13 @@ public class Ledger
 
     public void writeBlockFile (ref Block block) @safe
     {
-        auto fd = File("ledger.dat", "a"); // TODO move to config.yaml
+        auto dir = this.data_dir;
+        // Make sure we're in the right directory
+        if (!std.file.exists(dir) || !std.file.isDir(dir))
+        std.file.mkdir(dir);
```

Comment:
This should probably be moved somewhere in `main()` or in the `Node` constructor. The data directory should be created once, some time after parsing the `data_dir` config option.

Reply:
add param (ref File fd, ref Block block)